### ISSUE TITLE
IndentationCheck. Existance of comment affects violation message.

### ISSUE THREAD

**Issue by pirat9600q** [checkstyle/checkstyle#693]
Comment inside array initialization braces may affects violation message.
Checkstyle config:
```xml
<module name="Checker">
  <module name="TreeWalker">
    <module name="Indentation">
      <property name="arrayInitIndent" value="4"/>
      <property name="basicOffset" value="4"/>
      <property name="braceAdjustment" value="0"/>
      <property name="lineWrappingIndentation" value="4"/>
    </module>
  </module>
</module>
```
Input file:
```java
class ArrayInput { //indent:0 exp:0
    void m() { //indent:4 exp:4
        int[] array = {
           1,2,3 //indent:11 exp:12 warn
        }; //indent:8 exp:8
    } //indent:4 exp:4
} //indent:0 exp:0
```
Violation message:
`ArrayInput.java:4: 'array initialization' child have incorrect indentation level 11, expected level should be 12.`
Input file:
```java
class ArrayInput { //indent:0 exp:0
    void m() { //indent:4 exp:4
        int[] array = { //comment !
           1,2,3 //indent:11 exp:12 warn
        }; //indent:8 exp:8
    } //indent:4 exp:4
} //indent:0 exp:0
```
Violation message:
`ArrayInput.java:4: 'array initialization' child have incorrect indentation level 11, expected level should be one of the following: 12, 24, 26.`

**Comment by romani** [comment]
yes, message should not change.

**Comment by liscju** [comment]
Issue is involved with how ArrayInitHandler deals with children expected level. In ArrayInitHandler expected children level is calculated in ArrayInitHandler.getChildrenExpectedLevel as three different possibilities
a) current intend of Array Init line + lineWrappingIntendation(as usual)
b) if there is non blank character in line with Array Init,intendation in children can be as this character for example:
<pre>
int[] array = { //comment !
                1,2,3 //same intend as prev comment start, valid in this moment
}; 
</pre>

c) if there is non blank character in line with Array Init,intendation in children can be intend as this character + lineWrappingIntendation, for example:

<pre>
int[] array = { //comment !
                    1,2,3 //intend as start of comment + lineWrappingIntendation(4 here), valid in this moment
}; 
</pre>

Because of this behaviour, there is different expected level of child indentation. I do not know if this behaviour is expected(cases b and c), but if it is not ,its easy to fix this checking for non blank character that it is not start of comment , or in comment

**Comment by liscju** [comment]
I created a test case for this issue in commit 
https://github.com/liscju/checkstyle/commit/d1c40d50c69e6688c5a7e09318cf120d6fe2655f
and not found specified difference between violations. I dont know whether indentation comments (like //indent:8 exp:8) change anything in checking lines commented already like:
int[] array = { //comment ! //indent:8 exp:8
but if not, issue can be closed.

**Comment by romani** [comment]
@mkordas , can review that issue.

**Comment by mkordas** [comment]
@romani - issue raised by @pirat9600q is valid.

@liscju, in liscju@d1c40d5 you have one difference compared to original issue report:
```
int[] array = {
```
vs.
```
int[] array = { //indent:8 exp:8
```
Moving that comment reveals the issue.

@romani, what do you think? Should allowed indentation depend on comment placement?

**Comment by romani** [comment]
> Should allowed indentation depend on comment placement?

No. Comment is placed at the end of line, indentation is metric of line beginning.

**Comment by rnveach** [comment]
@romani 
> > Should allowed indentation depend on comment placement?
> 
> No.

So we should remove the placement based on the comment from the previous line?
In this example the violation should only say `expected level should be one of the following: 12, 26` regardless of comment or not, dropping off the 24.
Do you want default formatting for all IDEs?

**Comment by romani** [comment]
> So we should remove the special placement based on the comment from the previous line?
@rnveach ,

yes, trailing comments should not affect indentation.

> In this example the violation should only say expected level should be one of the following: 12, 26 regardless of comment or not, 

I do not know why `26` is required. Message should be the same as with comment and without comment.

> Do you want to see default formatting for all IDEs?

ok, lets see what IDEs do in such cases.
